### ISSUE TITLE
Starting Delayed Game from the Beginning

### ISSUE THREAD

**Issue by SlimJimPoisson** [kmac/mlbv#19]
I am most often in a situation where I want to download a game from the beginning that is currently in progress. In a perfect world my recording would begin from the opening of the pre-game show. With the tool it appears that I have two options. The first is to use "--from-start" option, but that means that it will download 1.5gb of "The event will begin shortly" screen which takes some times to complete on many days for me. The other option is to use "--inning 1", but that seems to start the recording somewhere during the second at bat in the top of the first which is sub-optimal.

Today I decided to test the "--inning-offset" option to see if I could create something better, but no matter what value I placed with inning-offset the recording started at the exact same spot. The documentation isn't very clear so I wasn't sure if the inning-offset worked with live games, or only with archived ones.

Anyway, in case it matters here is the text from two of my tests. As you can see the text a little different, but the recording was the same. Here is what I used:

[Command #1] mlbv --yesterday --fetch --resolution 720p --team atl --inning 1 --inning-offset 0
Retrieving inning info to locate 'top 1' inning
Found inning start: 2018-08-10T23:36:29.000Z
Live game: game start: 2018-08-10 23:35:00+00:00, inning start: 2018-08-10T23:36:29.000Z
Calculated live game negative inning offset (from now): 02:10:50

[Command #2] mlbv --yesterday --fetch --resolution 720p --team atl --inning 1 --inning-offset 500
Retrieving inning info to locate 'top 1' inning
Found inning start: 2018-08-10T23:36:29.000Z
Live game: game start: 2018-08-10 23:35:00+00:00, inning start: 2018-08-10T23:36:29.000Z
Calculated live game negative inning offset (from now): 02:15:07

As you can see the Calculated live game negative inning offset is different, but as I said the video start time was the same.

Am I approaching this correctly, or is there another method I could use to solve my problem?

**Comment by kmac** [comment]
If it's a live game, I find that starting with `--inning t1` or short-form `-i t1` brings me pretty much to the first pitch of the game - sometimes it's maybe a pitch or two behind, but it's pretty much bang on. 
That's because I'm keying it off of the first pitch event timestamp from the game data, and feeding that into streamlink. If it's starting from the second at-bat that doesn't match what I'm seeing. There may be a bug in there that I'm not seeing because of some sort of time zone issue (the calculation is a bit funky for live games). I'll have to take a look. I may get you to run with the --debug option which would save extra info like the json files, etc.

For archived games, I have no good solution that it is accurate (see #7). The data just isn't there, or I can't find it. I added the --inning-offset option in order to have some way of fiddling with this part of things. I must admit I don't think I tested it with a game that's in progress. The behaviour of streamlink is completely different for live stream vs archived games.

For now I think you're probably stuck with the --from-start option, or if it's an archived game then you don't even need that (it will start from the beginning).

**Comment by SlimJimPoisson** [comment]
I think I understand far more about the issues you are facing. Along the way I also think I located 1½ bugs.

In mlbv.py you have:

`   if args.inning_offset:`
`config.CONFIG.parser['stream_start_offset_secs'] = str(args.inning_offset)`

This leads to two problems. First if you set --inning-offset to 0 (as I did) in the command line then the If statement here fails and the argument rolls over to the default from config.py. No big deal (call it ½ a bug) as merely setting the offset to 1 will work and achieve almost the same result.

The bigger problem is that in stream.py references that variable differently in
`        offset_secs += config.CONFIG.parser.getint('stream_start_offset', 240)`
Here you call it 'stream_start_offset' not 'stream_start_offset_secs' so the variable name needs to be changed in one place or the other. I had wondered if using the '240' literal as a parameter was worth mentioning but my suspicion is that parameter doesn't mean anything as it will never be used.

I also added one line to the true side of the media_state == 'MEDIA_ON' statement:
`        offset_secs -= config.CONFIG.parser.getint('stream_start_offset_secs', 0)`
in order to have the argument applied to live games as well. Since we are offsetting from the end I changed the sign from + to - which makes sense to me, but I have not had the chance to test this code yet as I created it long after the live games had been finished. I will test it using tomorrow's games.

My hope is that I can add --inning-offset -600 and that should be enough to catch the pre-game intro in most cases, but will certainly catch the first pitch and eliminate most of the test pattern that precedes the live games.

**Comment by SlimJimPoisson** [comment]
My additional line worked perfectly. This issue is closed for me.

**Comment by kmac** [comment]
Hi, thanks for this!

There are three changes here. 
The first one I've handled by changing the code to check for 'is not None'. The second was definitely a bug, and the third one I've added the extra line you've given me.

I've gone ahead and translated the 3 points you made here into a new commit. If you wouldn't mind taking a look to confirm I'd appreciate it.

I won't do a full release yet since I think you're the only one affected.

**Comment by SlimJimPoisson** [comment]
Good thinking to be cautious on this one. I haven't tested anything, but even if it works for me it will become broken for everyone else.

The problem is that --inning-offset was designed to assist with archived games. It was not being applied to live games for good reason. You certainly wouldn't want the same default values for both (240 for archived and 0 for live makes sense, essentially what it was).

The only time it should make sense to use an offset with a live game is in my case where I want to record the pregame show (or in a perfect world starting from the same spot that the archived game would begin).

I have two suggestions:
1. Just ignore it. Roll back my change and I will just include this on my version.
2. Add the ability to use "--inning 0" where for live games it would use inning t1 and then subtract (or in this case add) a set number of seconds to the inning t1 time (600 should be plenty as a default) in hopes of getting the pre-game show as well. For archived games this option it would be the same as "--from-start". This would actually allow you to eliminate the "--from-start" option which was always incompatible with the --inning option (however, the system does not actually trap for that user error at the moment).

**Comment by kmac** [comment]
Yes this does break the existing functionality - something I noticed that I was suddenly not getting dropped in right near the first pitch.

I'm changing this to apply the negative offset only if `--inning-offset` (or the config file equivalent `stream_start_offset_secs`) is set to something other than the default value. So if you set --inning-offset on the command-line you should be able to adjust your start time as needed. 

I'm still unsure as to why you see this though. For me it's quite accurate, since the calculation is relatively straightforward for live games, and it's based on the timestamp of the first pitch.
